x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/aggregations/metrics/HistogramPercentileAggregationTests.java

```diff
@@ -103,28 +105,35 @@ public void testHDRHistogram() throws Exception {
         }
         client().admin().indices().refresh(new RefreshRequest("raw", "pre_agg")).get();
 
-        SearchResponse response = client().prepareSearch("raw").setTrackTotalHits(true).get();
-        assertEquals(numDocs, response.getHits().getTotalHits().value);
+        assertHitCount(client().prepareSearch("raw").setTrackTotalHits(true), numDocs);
 
-        response = client().prepareSearch("pre_agg").get();
-        assertEquals(numDocs / frq, response.getHits().getTotalHits().value);
+        assertHitCount(client().prepareSearch("pre_agg"), numDocs / frq);
 
         PercentilesAggregationBuilder builder = AggregationBuilders.percentiles("agg")
             .field("data")
             .method(PercentilesMethod.HDR)
             .numberOfSignificantValueDigits(numberOfSignificantValueDigits)
             .percentiles(10);
 
-        SearchResponse responseRaw = client().prepareSearch("raw").addAggregation(builder).get();
-        SearchResponse responsePreAgg = client().prepareSearch("pre_agg").addAggregation(builder).get();
-        SearchResponse responseBoth = client().prepareSearch("pre_agg", "raw").addAggregation(builder).get();
+        SearchResponse responseRaw = null;
+        SearchResponse responsePreAgg = null;
+        SearchResponse responseBoth = null;
+        try {
```

Comment:
We can nest `assertResponse` calls here like: 

```
 assertResponse(
            client().prepareSearch("raw").addAggregation(builder),
            responseRaw -> assertResponse(
                client().prepareSearch("pre_agg").addAggregation(builder),
                responsePreAgg -> assertResponse(client().prepareSearch("pre_agg", "raw").addAggregation(builder), responseBoth -> {
                    InternalHDRPercentiles percentilesRaw = responseRaw.getAggregations().get("agg");
                    InternalHDRPercentiles percentilesPreAgg = responsePreAgg.getAggregations().get("agg");
                    InternalHDRPercentiles percentilesBoth = responseBoth.getAggregations().get("agg");
                    for (int i = 1; i < 100; i++) {
                        assertEquals(percentilesRaw.percentile(i), percentilesPreAgg.percentile(i), 0.0);
                        assertEquals(percentilesRaw.percentile(i), percentilesBoth.percentile(i), 0.0);
                    }
                })
            )
        );
```

Reply:
Tbh opted to avoid multiple nesting calls to be more straightforward to read, but I can see that we also want to move the `decRef` responsibility outside of each test method whenever's possible. Updated.

Reply:
We should avoid calling decRef all over the tests as this makes very difficult to maintain. We should minimise as much as possible the places where we call that method.